### PR TITLE
Refactor to remove `_.escapeRegExp`

### PR DESCRIPTION
**Pull request by stephenwade**
> Which issue, if any, is this issue related to?

Ref #4412.

> Is there anything in the PR that needs further explanation?

No.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@stephenwade Thank you for creating this PR!

The `escape-string-regexp` package seems good, but I think it better if we could avoid adding a new dependency.
So, how about replacing 

```js
.replace(new RegExp(_.escapeRegExp('(' + message.rule + ')') + '$'), '')
```

with the following tiny function not using regex?

```js
/**
 * @param {string} text
 * @param {string} rule
 * @returns {string}
 */
const removeRuleSuffix = (text, rule) => {
	if (text.endsWith(rule)) {
		return text.slice(0, text.lastIndexOf(rule));
	}
	return text;
}

// usage
removeRuleSuffix(message.text.replace(/\.$/, ''), `(${message.rule})`);
```

**Comment by stephenwade** [comment]
@ybiquitous I have modified the code like you suggested to avoid using a regular expression.

File: lib/formatters/stringFormatter.js [hunk 1]
```diff
@@ -147,6 +146,27 @@ function formatter(messages, source) {
...
+
+		result = result
+			// Remove all control characters (newline, tab and etc)
+			.replace(/[\u0001-\u001A]+/g, ' ') // eslint-disable-line no-control-regex
+			.replace(/\.$/, '');
+
+		const ruleString = `(${message.rule})`;
+
+		if (result.endsWith(ruleString)) {
+			result = result.slice(0, result.lastIndexOf(ruleString));
```
Comment (ybiquitous) [inline lib/formatters/stringFormatter.js:164]: Thank you! In addition, how about adding `.trim()` as below?

```suggestion
			result = result.slice(0, result.lastIndexOf(ruleString)).trim();
```

```js
// input
'Unexpected invalid hex color "#ggg" (color-no-invalid-hex)'
//=> without trim
'Unexpected invalid hex color "#ggg" '
//=> with trim
'Unexpected invalid hex color "#ggg"'
```

Maybe, we may need more test cases of `stringFormatter.js` to verify this behavior change. 🤔

Comment (stephenwade) [reply lib/formatters/stringFormatter.js:164]: @ybiquitous I added a space to the `ruleString` that we attempt to remove. I also added a test for the behavior.

```diff
-		const ruleString = `(${message.rule})`;
+		const ruleString = ` (${message.rule})`;
```

What do you think?

Comment (ybiquitous) [reply lib/formatters/stringFormatter.js:164]: It also looks good! 👍🏼

**Review by ybiquitous — APPROVED** [review]
@stephenwade Thank you so much! LGTM 👍🏼

**Pull request merged**